parlai/core/metrics.py

```diff
@@ -258,6 +258,9 @@ def report(self):
             for k in self.metrics_list:
                 if self.metrics[k + '_cnt'] > 0 and k != 'correct' and k != 'f1':
                     m[k] = round_sigfigs(self.metrics[k] / self.metrics[k + '_cnt'], 4)
+        # if 'loss' in m and 'ppl' not in m:
+        if 'loss' in m:
+            m['ppl'] = round_sigfigs(math.exp(m['loss']), 4)
```

Comment:
this doesn't make sense in that loss could mean anything (0/1, ranking, cross ent, mse).. this should only be for a specific kind of loss

Reply:
yeah I was a little worried about that too, hmm I think I found a better way to do this